test/tests/interfaces/pbs_sched_interface_test.py

```diff
@@ -0,0 +1,207 @@
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15007) returned from server' in e.msg[1])
+
+        try:
+            self.server.manager(MGR_CMD_CREATE,
+                                SCHED,
+                                id="testCreateSched",
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.fail(e.msg)
+
+        # Check for delete permission
+        self.server.manager(MGR_CMD_CREATE,
+                            SCHED,
+                            id="testDeleteSched")
+        try:
+            self.server.manager(MGR_CMD_DELETE,
+                                SCHED,
+                                id="testDeleteSched",
+                                runas=OPER_USER)
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15007) returned from server' in e.msg[1])
+
+        try:
+            self.server.manager(MGR_CMD_DELETE,
+                                SCHED,
+                                id="testDeleteSched",
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.fail(e.msg)
+
+        # Check for attribute set permission
+        try:
+            self.server.manager(MGR_CMD_SET,
+                                SCHED,
+                                {'sched_cycle_length': 12000},
+                                id="TestCommonSched",
+                                runas=OPER_USER)
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15007) returned from server' in e.msg[1])
+
+        try:
+            self.server.manager(MGR_CMD_SET,
+                                SCHED,
+                                {'sched_cycle_length': 12000},
+                                id="TestCommonSched",
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.fail(e.msg)
+
+    def test_delete_default_sched(self):
+        """
+        Delete default scheduler.
+        """
+        try:
+            self.server.manager(MGR_CMD_DELETE,
+                                SCHED,
+                                id="default")
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15214) returned from server' in e.msg[1])
+
+    def test_set_sched_attributes(self):
+        """
+        Set an attribute of a scheduler object .
+        """
+        try:
+            # Default values will be set
+            self.server.manager(MGR_CMD_SET,
+                                SCHED,
+                                {'sched_cycle_length': 1234},
+                                id="TestCommonSched")
```

Comment:
It the test is ran from user XYZ, then XYZ user become MGR and all commands are executed with that user. So no need to add runas parameter.